src/ros_answers_miner/models.py

```diff
@@ -5,27 +5,151 @@
 """
 __all__ = ('Answer', 'Comment', 'Question', 'User')
 
-from typing import AbstractSet
+from sortedcollections import OrderedSet
 
-import attr
+class User:
+    url: str
 
+    def __init__(self, url: str):
+        self.url = url
 
-@attr.s(auto_attribs=True)
-class Question:
-    url: str
-    tags: AbstractSet[str]
+    def __key(self):
+        return (self.url)
+
+    def __hash__(self):
+        return hash(self.__key())
+
+    def __eq__(self, other):
+        return isinstance(other, User) and self.__key() == other.__key()
+
+    def to_json(self):
+        return {
+            '__class__': 'User',
+            'url': self.url
+        }
+
+
+class Comment:
+    date: str
+    votes: int
+    content: str
+
+    user: User
+
+    def __init__(self, date, votes, content, user):
+        self.date = date
+        self.votes = votes
+        self.content = content
+        self.user = user
+
+    def __key(self):
+        return (self.votes, self.content)
+
+    def __hash__(self):
+        return hash(self.__key())
+
+    def __eq__(self, other):
+        return isinstance(other, Comment) and self.__key() == other.__key()
+
+    def to_json(self):
+        return {
+            '__class__': 'Comment',
+            'date': str(self.date),
+            'votes': self.votes,
+            'content': self.content,
+            'user': self.user.to_json()
+        }
 
 
-@attr.s(auto_attribs=True)
 class Answer:
     accepted: bool
 
+    date: str
+    votes: int
+    user: User
 
-@attr.s(auto_attribs=True)
-class Comment:
-    contents: str
+    content: str
+    comments: OrderedSet[Comment]
 
+    def __init__(self, accepted, date, votes, user, content, comments):
+        self.accepted = accepted
+        self.date = date
+        self.votes = votes
+        self.user = user
+        self.content = content
+        self.comments = comments
 
-@attr.s(auto_attribs=True, frozen=True)
-class User:
+    def __key(self):
+        return (self.accepted, self.votes, self.content)
+
+    def __hash__(self):
+        return hash(self.__key())
+
+    def __eq__(self, other):
+        return isinstance(other, Answer) and self.__key() == other.__key()
+
+    def to_json(self):
+        return {
+            '__class__': 'Answer',
+            'accepted': self.accepted,
+            'date': str(self.date),
+            'votes': self.votes,
+            'user': self.user.to_json(),
+            'content': self.content,
+            'comments': OrderedSet([comment.to_json() for comment in self.comments])
+        }
+
+
+class Question:
+    # Default information about a question
     url: str
+    date: str
+    votes: int
+    views: int
+    user: User
+
+    # Content of the question
+    title: str
+    content: str
+
+    # Tags, comments and answers
+    tags: OrderedSet[str]
+    comments: OrderedSet[Comment]
+    answers: OrderedSet[Answer]
+
+    def __init__(self, url, date, votes, views, user, title, content, tags, comments, answers):
```

Comment:
Again, this is why attrs exists! Don't roll your own __init__, __hash__, __cmp__, __eq__, etc.